plugins/poll/app/models/poll.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+class Poll < ActiveRecord::Base
+  # because we want to use the 'type' column and don't want to use STI
+  self.inheritance_column = nil
+
+  belongs_to :post
+
+  has_many :poll_options, dependent: :destroy
+  has_many :poll_votes, dependent: :delete_all
+
+  validates :type, inclusion: { in: %w(regular multiple number) }
+  validates :status, inclusion: { in: %w(open closed) }
+  validates :results, inclusion: { in: %w(always vote closed) }
```

Comment:
Perhaps we can store the strings above in an enum so that we only have to store an integer in the columns?

Reply:
I have grown to hate integer enums. I feel the storage space "gained" is not worth the clarity of

`SELECT * FROM polls WHERE status = 'closed'` 

Which is IMHO 10000x better than 

`SELECT * FROM polls WHERE status = 0`

But I sure can create SQL types to enforce the values on these columns.

Reply:
A single digit integer takes up 2 bytes while the shortest string here takes up 5 bytes. The longest string in this case takes up 9 bytes. On top of that, we don't know if we'll eventually introduce `type/status/results` that are longer in length. The other benefit is that indexing an integer column is usually faster and smaller in size. The gains are quite significant IMO and is a tradeoff that I think we should take and has already taken in other places of our app. 

On the Ruby side of things, using an enum allows us to avoid specifying a common string in multiple places. I really like the pattern of `Poll.types[:regular]` or `Poll.status[:closed]` so that we only need to declare the string once and avoid any possibility of making a typo in multiple places.

Reply:
Sure, feel free to do the changes ;)